src/components/UpdateModal.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+
+import React, { Component } from 'react'
+import { View, StyleSheet } from 'react-native-web'
+import Text from './Text'
+import Button from './Form/Button'
+import Modal from './Modal'
+import { BASIC_SPACING } from '../styles'
+
+const shell = window.require('electron').shell
+
+type Props = {
+  onError?: () => void,
+  onDismiss?: () => void,
+}
+
+type State = {
+  open: boolean,
+  update: {
+    title?: string,
+    url?: string,
+    description?: string,
+    button?: string,
+  },
+}
+
+const CURRENT_VERSION = '0.4.2'
+const VERSION_URL = 'https://mainframe.com/onyx-version.json'
+
+export default class UpdateModal extends Component<Props, State> {
+  state: State = {
+    open: false,
+    update: {
+      title: 'New version',
+      description: 'A new version of Onyx was released',
+      button: 'Download',
+    },
+  }
+
+  componentDidMount() {
+    fetch(VERSION_URL)
+      .then(response => {
+        if (response.ok) {
+          return response.json()
+        } else {
+          this.props.onError && this.props.onError()
+        }
+      })
+      .then(data => {
+        if (data.version && data.version > CURRENT_VERSION) {
```

Comment:
`version` is a string, so using `>` for comparison will so a string comparison, not a number one, better simply use `===` here.